app/lib/three_scale/json_representer.rb

```diff
@@ -22,7 +22,7 @@ def to_node(options)
     doc = options.fetch(:doc)
     xml = to_xml(builder: builder)
 
-    Nokogiri::XML::Node.new('root', doc).replace(xml)
+    Nokogiri::XML::Node.new('root', doc).add_child(xml)
```

Comment:
I am a little worry this may change the behavior and I don't see any tests related to JSONRepresenter. If I am not mistaken this class manages the JSON format response for models such as https://github.com/3scale/porta/blob/25fa9bc53bdd3eeb8f2a9c8e882f771c85c3c3cb/app/representers/application_plans_representer.rb#L1-L7

Maybe it is covered by integration tests... I am not sure. Have you tested this manually by calling some 3scale API endpoints?

Reply:
Can you also explain this change? Did API change or what is the point?

Reply:
Can whatever issue happened be explained by this change in nokogiri?

> The Node methods add_previous_sibling, previous=, before, add_next_sibling, next=, after, replace, and swap now correctly use their parent as the context node for parsing markup. These methods now also raise a RuntimeError if they are called on a node with no parent. [rubys/nokogumbo#160]

Reply:
@josemigallas I have done some manual tests with 3scale API endpoints and they are working.
@akostadinov the way `Nokogiri::XML::Node#new` works has changed and now when introducing the xml we need to have a parent node (created at `#new`). What `#add_child` does is introducing the xml code inside that parent. I have checked with the console when `bundle exec rails t test/integration/user-management-api/buyers_users_test.rb:476`:

You can check the differences here: [https://diffonline.net/yPYv9bxyVP](https://diffonline.net/yPYv9bxyVP)

Reply:
JSONRepresenter is used by many other representers:
```
$ rg JSONRepresenter
test/test_helpers/represented_api_routing.rb
27:    include ThreeScale::JSONRepresenter
35:    include ThreeScale::JSONRepresenter
44:    include ThreeScale::JSONRepresenter
52:    include ThreeScale::JSONRepresenter

app/representers/access_token_representer.rb
4:  include ThreeScale::JSONRepresenter

app/representers/account_plan_representer.rb
2:  include ThreeScale::JSONRepresenter
...
```

`to_node` is probably used for XML representing.

@lvillen , what I didn't understand is why `replace` stopped working. `#add_child` I assume would be equivalent to `#replace` when same element does not exist anymore. But is it possible same element to exist? Is it possible to make `#replace` work?

Perhaps it is fine to have it like it is though. Still it is preferable to understand :)